src/main/java/io/vertx/core/eventbus/impl/ReplyHandler.java

```diff
@@ -45,7 +45,8 @@
 
   private void trace(Object reply, Throwable failure) {
     VertxTracer tracer = context.tracer();
-    if (tracer != null && src) {
+    Object trace = this.trace;
```

Comment:
why this new Object? thread-safety concerns?

Reply:
yes, reading a ref and then using it